test/test_basic_extractor.py

```diff
@@ -0,0 +1,171 @@
+"""
+Test the code in rnnalayse.extractors.base_extractor.py.
+"""
+
+import itertools
+import unittest
+from unittest.mock import patch, MagicMock
+
+import numpy as np
+from overrides import overrides
+import torch
+
+from rnnalyse.extractors.base_extractor import Extractor
+from rnnalyse.models.language_model import LanguageModel
+from test_utils import create_sentence_dummy_activations
+
+
+# GLOBALS
+ACTIVATION_DIM = 10
+ACTIVATION_NAMES = [(0, "hx"), (0, "cx")]
+ACTIVATIONS_DIR = "test_data"
+
+
+class MockLanguageModel(LanguageModel):
```

Comment:
I think this could also be directly mocked by decorating the function where it is called with mock.patch or mock.patch.object. This is also a nice solution though.

Reply:
The way I have the model produce these dummy activations in an ordered manner makes this unfortunately very tricky (when mocking the forward function, I need access to `self.all_pairs`. After one test when this list was consumed by `next(...)` I want to reset it for the next test, but when using Mocks I have no (easy) way (I found) to access this variable again). Therefore I opted for a custom mock class in this case.